packages/orchestration/src/exos/portfolio-holder-kit.js

```diff
@@ -0,0 +1,176 @@
+import { M, mustMatch } from '@endo/patterns';
+import { E } from '@endo/far';
+import { Fail } from '@endo/errors';
+import {
+  TopicsRecordShape,
+  PublicTopicShape,
+} from '@agoric/zoe/src/contractSupport/topics.js';
+import { makeScalarBigMapStore } from '@agoric/vat-data';
+import { Shape as NetworkShape } from '@agoric/network';
+import { VowShape } from '@agoric/vow';
+
+const { fromEntries } = Object;
+const { Vow$ } = NetworkShape;
```

Comment:
these each return a Vow or not. Please do not use this `M.or` loosener. It gives a false sense of what will be validated.

If you want to include documentation about the type of the unwrapped vow, jsdoc suffices